### PR TITLE
Add compound components exercise

### PR DESCRIPTION
**Pull request by sapegin**

### CODE REVIEW

File: src/exercises/2-3-Compound_components/templates/Dialog.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  DialogCloseButton
+} from './final/Dialog';
+const [name, setName] = React.useState('');
+const [draftName, setDraftName] = React.useState('');
+const [isVisible, setIsVisible] = React.useState(false);
+const onClose = () => setIsVisible(false);
+<>
+  {isVisible && (
+    <Dialog>
+      <DialogBackdrop onClose={onClose} />
```
Comment (okonet) [inline src/exercises/2-3-Compound_components/templates/Dialog.md:44]: You mean `onClick={onClose}`?

File: src/exercises/2-3-Compound_components/templates/Dialog.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      <DialogContent>
+        <DialogCloseButton onClose={onClose} />
```
Comment (okonet) [inline src/exercises/2-3-Compound_components/templates/Dialog.md:46]: Same?

File: src/exercises/2-3-Compound_components/templates/Dialog.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  DialogBackdrop,
+  DialogHeader,
+  DialogBody,
+  DialogFooter,
+  DialogCloseButton
+} from './final/Dialog';
+const [name, setName] = React.useState('');
+const [draftName, setDraftName] = React.useState('');
+const [isVisible, setIsVisible] = React.useState(false);
+const onClose = () => setIsVisible(false);
```
Comment (okonet) [inline src/exercises/2-3-Compound_components/templates/Dialog.md:40]:
```suggestion
const closeDialog = () => setIsVisible(false);
```

File: src/exercises/2-3-Compound_components/templates/Dialog.md (hunk 1 above)
Comment (okonet) [inline src/exercises/2-3-Compound_components/templates/Dialog.md:44]:
```suggestion
      <DialogBackdrop onClick={closeDialog} />
```

File: src/exercises/2-3-Compound_components/templates/Dialog.md (hunk 2 above)
Comment (okonet) [inline src/exercises/2-3-Compound_components/templates/Dialog.md:46]:
```suggestion
        <DialogCloseButton onClick={closeDialog} />
```

File: src/exercises/2-3-Compound_components/templates/Dialog.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+              <br />
+              <input
+                type="text"
+                value={draftName}
+                onChange={event => setDraftName(event.target.value)}
+              />
+            </label>
+          </DialogBody>
+          <DialogFooter>
+            <button type="button" onClick={onClose}>
```
Comment (okonet) [inline src/exercises/2-3-Compound_components/templates/Dialog.md:67]:
```suggestion
            <button type="button" onClick={closeDialog}>
```

File: src/exercises/2-3-Compound_components/templates/Dialog.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  {isVisible && (
+    <Dialog>
+      <DialogBackdrop onClose={onClose} />
+      <DialogContent>
+        <DialogCloseButton onClose={onClose} />
+        <form
+          onSubmit={event => {
+            event.preventDefault();
+            setName(draftName);
+            onClose();
```
Comment (okonet) [inline src/exercises/2-3-Compound_components/templates/Dialog.md:51]:
```suggestion
            closeDialog();
```

**Comment by sapegin** [comment]
These names will make more sense in the next exercise — this would be the hook API, so let's keep them for now, and if you still disagree, we can change them after you see the next exercise 🦆

**Review by okonet — APPROVED** [review]

**Pull request merged**